packages/shared/src/components/notifications/EnableNotification.tsx

```diff
@@ -53,16 +58,30 @@ function EnableNotification({
 
   const onEnable = async () => {
     const permission = await onTogglePermission();
+
+    if (permission === null) {
+      return;
+    }
+
     const isGranted = permission === 'granted';
 
-    setIsEnabled(isGranted);
+    onHasEnabledPermission(isGranted);
   };
 
+  const hasEnabled = (hasPermission || hasPermissionCache) && isEnabled;
+
+  useEffect(
+    () => () => {
+      onHasEnabledPermission(false);
+    },
+    [],
+  );
```

Comment:
This is a bit confusing, we should favor the `return` statement in this case to make it clearer

Reply:
Ah, I totally agree with that 🔭